sparse-checkout.c

```diff
@@ -0,0 +1,37 @@
+#include "cache.h"
+#include "config.h"
+#include "dir.h"
+#include "hashmap.h"
+#include "repository.h"
+#include "run-command.h"
+#include "sparse-checkout.h"
+#include "strbuf.h"
+
+static int core_sparse_checkout = -1;
```

Comment:
This file may not actually be important, since a lot of the logic is going into `dir.c`.